File: frontend/src/app/models/tournament/tournament.response.ts
import { UserRef } from "../common/user-ref.model";
import { TournamentItemResponse } from "./tournament-item.response";

export interface TournamentResponse extends TournamentItemResponse {
    referees: UserRef[];
    scoringRules: Record<string, number>;
    qrCodeGenerated: boolean;
}

File: src/main/java/com/tournament/app/footycup/backend/dto/tournament/TournamentResponse.java
package com.tournament.app.footycup.backend.dto.tournament;

import com.tournament.app.footycup.backend.dto.common.UserRef;

import java.time.LocalDate;
import java.util.List;
import java.util.Map;

public record TournamentResponse(
        Long id,
        String name,
        LocalDate startDate,
        LocalDate endDate,
        String location,
        String status,
        String system,
        UserRef organizer,
        boolean publicVisible,
        boolean followed,
        boolean qrCodeGenerated,
        List<UserRef> referees,
        Map<String, Integer> scoringRules
) {
}
